### PR TITLE
I 03

### PR DESCRIPTION
**Pull request by venividiviciregina**

### CODE REVIEW

File: src/test/java/com/school/project/ProjectApplicationTests.java [hunk 1]
```diff
@@ -13,4 +13,4 @@
     public void contextLoads() {
```
Comment (MichelRogov) [inline src/test/java/com/school/project/ProjectApplicationTests.java]: shouldn't be in PR

File: src/main/java/com/school/project/exception/UserNotFoundException.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.school.project.exception;
+
+public class UserNotFoundException extends RuntimeException {
+
+    public UserNotFoundException(String msg) {
+        super(msg);
+    }
+
+<<<<<<< HEAD
```
Comment (sergeylukichev) [inline src/main/java/com/school/project/exception/UserNotFoundException.java]: I doubt the project is compilable

File: src/main/java/com/school/project/service/impl/UserAccountServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserAccountServiceImpl implements UserAccountService {
+    @Autowired
+    UserAccountRepository userAccountRepository;
+    @Override
+    public UserAccount createUserAccount(User user, AccountType accountType) {
+        UserAccount userAccount=UserAccount.builder()
+                .user((org.apache.catalina.User) user)
```
Comment (sergeylukichev) [inline src/main/java/com/school/project/service/impl/UserAccountServiceImpl.java]: What is this? (`org.apache.catalina.User`).

File: src/main/resources/dbmigration/V6__STUDENT.sql [hunk 4]
```diff
@@ -0,0 +1,3 @@
+INSERT INTO STUDENT(ID, NAME);
```
Comment (sergeylukichev) [inline src/main/resources/dbmigration/V6__STUDENT.sql]: I would create a separate PR for the SQL files only and then the second one for the controllers/repositories/business logic.

File: src/main/java/com/school/project/controller/UserController.java [hunk 5]
```diff
@@ -1,9 +1,60 @@
...
+@RequestMapping
 public class UserController {
+    @Autowired
+    ModelMapper modelMapper;
+    @Autowired
+    UserService userService;
+    @Autowired
+    UserRepository userRepository;
+
+    @PostMapping("/users")
```
Comment (sergeylukichev) [inline src/main/java/com/school/project/controller/UserController.java]: This class is already in another (still open) pull request. As long as another one is changed and merged, you will get a merge conflict.

File: src/main/java/com/school/project/service/impl/UserAccountServiceImpl.java (hunk 3 above)
Comment (MichelRogov) [reply src/main/java/com/school/project/service/impl/UserAccountServiceImpl.java]: will be rebase on develop after #13  will go in

File: src/main/resources/application.properties [hunk 6]
```diff
@@ -1,13 +1,23 @@
...
-spring.datasource.password=12345678
\ No newline at end of file
+spring.datasource.username=db_test
+spring.datasource.password=12345
+#flyway
+spring.flyway.locations=classpath:db/migration
+spring.flyway.baseline-on-migrate=true
+spring.flyway.validate-on-migrate=true
+spring.flyway.clean-on-validation-error=true
+spring.flyway.user=db_test
```
Comment (MichelRogov) [inline src/main/resources/application.properties]: why DB is called test here?

File: src/main/resources/application.properties [hunk 7]
```diff
@@ -1,13 +1,23 @@
...
 spring.jpa.properties.format_sql=true
 spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.MySQL5Dialect
 spring.jpa.database-platform=org.hibernate.dialect.MySQL5Dialect
 spring.jpa.hibernate.naming.physical-strategy=org.hibernate.boot.model.naming.PhysicalNamingStrategyStandardImpl
-
+#datasource
 spring.datasource.url=jdbc:mysql://localhost:3306/accounting?createDatabaseIfNotExist=true&useUnicode=true&useJDBCCompliantTimezoneShift=true&useLegacyDatetimeCode=false&serverTimezone=UTC
 spring.datasource.driverClassName=com.mysql.cj.jdbc.Driver
-spring.datasource.username=root
-spring.datasource.password=12345678
```
Comment (MichelRogov) [inline src/main/resources/application.properties]: please return previous two lines back

File: src/main/resources/application.properties [hunk 8]
```diff
@@ -1,13 +1,23 @@
...
+spring.datasource.username=db_test
+spring.datasource.password=12345
+#flyway
+spring.flyway.locations=classpath:db/migration
+spring.flyway.baseline-on-migrate=true
+spring.flyway.validate-on-migrate=true
+spring.flyway.clean-on-validation-error=true
+spring.flyway.user=db_test
+spring.flyway.password=12345
+spring.flyway.schemas=accounting
```
Comment (MichelRogov) [inline src/main/resources/application.properties]: AFAIK we don't need schema name, user, and password to make flyway working. The link below should go away as well

File: src/main/resources/data.sql [hunk 9]
```diff
@@ -0,0 +1,10 @@
+create table USER
```
Comment (MichelRogov) [inline src/main/resources/data.sql]: why do we have data.sql here? we are using flyway instead, aren't we?

File: src/main/resources/dbmigration/V1__USER_ACCOUNT_TABLE.sql [hunk 10]
```diff
@@ -0,0 +1,10 @@
+create table USER_ACCOUNT
```
Comment (MichelRogov) [inline src/main/resources/dbmigration/V1__USER_ACCOUNT_TABLE.sql]: please move this sql script to V1_00__INITIALIZE.sql
and add constraints to user_id
and delete suffix table from file name

File: src/main/resources/dbmigration/V2__USER.sql [hunk 11]
```diff
@@ -0,0 +1,8 @@
+INSERT INTO USER(ID, NAME);
```
Comment (MichelRogov) [inline src/main/resources/dbmigration/V2__USER.sql]: why do we have two scripts for user?? V1_01__USER.sql and V2__USER.sql

File: src/main/resources/dbmigration/V3__ACCOUNT_TYPE_TABLE.sql [hunk 12]
```diff
@@ -0,0 +1,10 @@
+create table ACCOUNT_TYPE
```
Comment (MichelRogov) [inline src/main/resources/dbmigration/V3__ACCOUNT_TYPE_TABLE.sql]: delete suffix table from file name

File: src/main/resources/dbmigration/V6__STUDENT.sql (hunk 4 above)
Comment (MichelRogov) [reply src/main/resources/dbmigration/V6__STUDENT.sql]: more than that, by our UML diagram we won't have such tables at all.

File: src/main/resources/dbmigration/V5__ADMIN.sql [hunk 13]
```diff
@@ -0,0 +1,3 @@
+INSERT INTO ADMIN(ID, NAME);
```
Comment (MichelRogov) [inline src/main/resources/dbmigration/V5__ADMIN.sql]: please delete this file, we won't have such table

File: src/main/resources/dbmigration/V4__TEACHER.sql [hunk 14]
```diff
@@ -0,0 +1,3 @@
+INSERT INTO TEACHER(ID, NAME);
```
Comment (MichelRogov) [inline src/main/resources/dbmigration/V4__TEACHER.sql]: please delete this file, we won't have such table

File: pom.xml [hunk 15]
```diff
@@ -44,7 +49,7 @@
         <dependency>
             <groupId>mysql</groupId>
             <artifactId>mysql-connector-java</artifactId>
-            <version>5.1.47</version>
+            <version>8.0.13</version>
```
Comment (MichelRogov) [inline pom.xml]: this should 5.1.47, pls change

File: src/main/resources/application.properties [hunk 16]
```diff
@@ -1,13 +1,19 @@
...
 #JPA
 spring.jpa.hibernate.ddl-auto=create
 spring.jpa.properties.format_sql=true
 spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.MySQL5Dialect
 spring.jpa.database-platform=org.hibernate.dialect.MySQL5Dialect
 spring.jpa.hibernate.naming.physical-strategy=org.hibernate.boot.model.naming.PhysicalNamingStrategyStandardImpl
-
+#datasource
 spring.datasource.url=jdbc:mysql://localhost:3306/accounting?createDatabaseIfNotExist=true&useUnicode=true&useJDBCCompliantTimezoneShift=true&useLegacyDatetimeCode=false&serverTimezone=UTC
-spring.datasource.driverClassName=com.mysql.jdbc.Driver
```
Comment (MichelRogov) [inline src/main/resources/application.properties]: please recover this line

**Comment by Koblov** [comment]
Приветствую Михаил,
1) In pom.xml <https://github.com/MichelRogov/Accounting/pull/18#discussion_r290226087>
 у меня и стоит версия 8.0.13


2) после ввернутой In src/main/resources/application.properties <https://github.com/MichelRogov/Accounting/pull/18#discussion_r290226314>:
выдал ошибку


Mit freundlichen Grüßen/ С Уважением
Viacheslav Koblov 
Brittendorfer Weg 19,14167 Berlin
Mobil: +49 176 555 623 84
http://immobilien.net.ru
http://vk.com/nedvizimost_germany

File: pom.xml [hunk 17]
```diff
@@ -36,6 +36,11 @@
             <artifactId>spring-boot-devtools</artifactId>
             <scope>runtime</scope>
         </dependency>
+        <dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <scope>runtime</scope>
```
Comment (MichelRogov) [inline pom.xml]: again: here should be version 5.1.47

File: src/main/resources/db/migration/V1_0__INITIALIZE.sql [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+alter table USER
+    add constraint ID unique (ID);
+
+alter table USER_ACCOUNT
+    add constraint ID unique (ID);
+alter table  USER_ACCOUNT
+    add constraint USER_ACCOUNT_USER_ID_USER_ID
+        foreign key (USER_ID) references USER(ID) ON DELETE CASCADE;
+
+
```
Comment (MichelRogov) [inline src/main/resources/db/migration/V1_0__INITIALIZE.sql]: get rid of these empty lines. after the last brace should be only one new line

File: src/main/resources/db/migration/V1_2__USERACCOUNT.sql [hunk 19]
```diff
@@ -0,0 +1,5 @@
+INSERT into USER_ACCOUNT(ID,USER_ID,REGISTRATION_DATE,ACCOUNT_ROLE) VALUES
+(1,1,'2019-05-04','STUDENT'),
+(2,2,'2019-05-04','STUDENT'),
+(3,2,'2019-05-04','STUDENT'),
+(4,4,'2019-05-04','STUDENT');
```
Comment (MichelRogov) [inline src/main/resources/db/migration/V1_2__USERACCOUNT.sql]: add one teacher pls.

**Comment by MichelRogov** [comment]
> Приветствую Михаил, 1) In pom.xml <[#18 (comment)](https://github.com/MichelRogov/Accounting/pull/18#discussion_r290226087)> у меня и стоит версия 8.0.13 2) после ввернутой In src/main/resources/application.properties <[#18 (comment)](https://github.com/MichelRogov/Accounting/pull/18#discussion_r290226314)>: выдал ошибку Mit freundlichen Grüßen/ С Уважением Viacheslav Koblov Brittendorfer Weg 19,14167 Berlin Mobil: +49 176 555 623 84 http://immobilien.net.ru http://vk.com/nedvizimost_germany

какую ошибку? нужен скрин и описание шагов как получить эту ошибку.

File: src/main/resources/application.properties [hunk 20]
```diff
@@ -1,13 +1,19 @@
...
+spring.datasource.driverClassName=com.mysql.cj.jdbc.Driver
```
Comment (MichelRogov) [inline src/main/resources/application.properties]: `spring.datasource.driverClassName=com.mysql.cj.jdbc.Driver` replace with `spring.datasource.driverClassName=com.mysql.jdbc.Driver`

**Pull request merged**